DNN Platform/Modules/DDRMenu/MenuBase.cs

```diff
@@ -15,48 +15,68 @@ namespace DotNetNuke.Web.DDRMenu
     using System.Xml;
     using System.Xml.Serialization;
 
+    using DotNetNuke.Abstractions.Portals;
     using DotNetNuke.Common.Utilities;
     using DotNetNuke.Entities.Host;
     using DotNetNuke.Entities.Portals;
     using DotNetNuke.Entities.Tabs;
     using DotNetNuke.Entities.Users;
     using DotNetNuke.Security.Permissions;
-    using DotNetNuke.UI;
     using DotNetNuke.Web.DDRMenu.DNNCommon;
     using DotNetNuke.Web.DDRMenu.Localisation;
     using DotNetNuke.Web.DDRMenu.TemplateEngine;
 
+    /// <summary>
+    /// Base class for multiple DDR Menu classes.
+    /// </summary>
     public class MenuBase
     {
         private readonly Dictionary<string, string> nodeSelectorAliases = new Dictionary<string, string>
-                                                                          {
-                                                                            { "rootonly", "*,0,0" },
-                                                                            { "rootchildren", "+0" },
-                                                                            { "currentchildren", "." },
-                                                                          };
+        {
+            { "rootonly", "*,0,0" },
+            { "rootchildren", "+0" },
+            { "currentchildren", "." },
+        };
 
         private Settings menuSettings;
 
         private HttpContext currentContext;
 
-        private PortalSettings hostPortalSettings;
+        private IPortalSettings hostPortalSettings;
 
+        /// <summary>
+        /// Gets or sets the template definition.
+        /// </summary>
         public TemplateDefinition TemplateDef { get; set; }
 
+        /// <summary>
+        /// Gets the portal settings for the current portal.
+        /// </summary>
         internal PortalSettings HostPortalSettings
         {
-            get { return this.hostPortalSettings ?? (this.hostPortalSettings = PortalController.Instance.GetCurrentPortalSettings()); }
+            get { return (PortalSettings)(this.hostPortalSettings ?? (this.hostPortalSettings = PortalController.Instance.GetCurrentSettings())); }

```

Comment:
Rather than casting, I think this should be marked as `Obsolete`, return `PortalController.Instance.GetCurrentPortalSettings()`, and use a `#pragma` to avoid the warning about using an obsolete member.

Reply:
Does this make sense ? https://github.com/dnnsoftware/Dnn.Platform/pull/4224/commits/45da40c51e44f89fefa2c67929141f5861054139

Reply:
Yup.  I looked through this and didn't find anything to change (except for some typos 🙂)